test/runnable/functype.d

```diff
@@ -296,6 +296,26 @@ void test10734()
 
 }
 
+void test14656()
+{
+    void unaryFun()(auto int a) pure nothrow @safe @nogc {}
```

Comment:
Currently applying only `auto` attribute on template function parameters does not make any errors, but I think it's incorrect.

This line should raise `Error: auto can only be used for template function parameters` as same as non-template functions.